src/main/java/it/geosolutions/httpproxy/HTTPProxy.java

```diff
@@ -209,12 +209,18 @@ public HttpRoute determineRoute(
                     LOGGER.info("Returning proxy route");
                     if (target.getSchemeName().equals("http")) {
                         LOGGER.debug("Setting http scheme");
-                        return new HttpRoute(target, null, httpHost,
```

Comment:
The code nesting here is a lot. I would refact this code a bit to provide a separate method doing the creation of the right HttpRoute object or I would turn the anonymous class into a nested one.

Reply:
created a new method for proxy HTTPRoute object creation